addons/docs/src/blocks/Source.tsx

```diff
@@ -32,13 +58,45 @@ type NoneProps = CommonProps;
 
 type SourceProps = SingleSourceProps | MultiSourceProps | CodeProps | NoneProps;
 
+const getSnippet = (
+  storyId: StoryId,
+  sourceContext: SourceContextProps,
+  docsContext: DocsContextProps
+): string => {
+  const { sources } = sourceContext;
+  const { storyStore } = docsContext;
+
+  const snippet = sources && sources[storyId];
+  const data = storyStore?.fromId(storyId);
```

Comment:
Don't we have parameters on docs context? It seems odd to reach into the store

Reply:
Parameters are only for the currently selected story but this could be ANY story

Reply:
Got it